packages/web-components/src/components/image-with-caption/image-with-caption.ts

```diff
@@ -0,0 +1,153 @@
+class DDSImageWithCaption extends ModalRenderMixin(LitElement) {
+  /**
+   * `true` handles re-opening after model is closed
+   * @private
+   */
+  private _handleClick() {
+    this.open = true;
+  }
+
+  /**
+   * The handler of `${ddsPrefix}-modal-closed` event from `<dds-modal>`.
+   */
+  private _handleCloseModal = () => {
+    this.open = false;
+  };
+
+  /**
+   * The handle for the listener of `${ddsPrefix}-modal-closed` event.
+   */
+  private _hCloseModal: Handle | null = null;
+
+  /**
+   * The lightbox.
+   */
+  @property({ type: Boolean, reflect: true })
+  lightbox = false;
+
+  /**
+   * The image source.
+   */
+  @property({ reflect: true, attribute: 'default-src' })
+  defaultSrc = '';
+
+  /**
+   * The alt text.
+   */
+  @property({ reflect: true })
+  alt = '';
+
+  /**
+   * The heading.
+   */
+  @property({ reflect: true })
+  heading = '';
+
+  /**
+   * The description.
+   */
+  @property({ reflect: true })
+  copy = '';
+
+  /**
+   * `true` if the modal should be open.
+   */
+  @property({ type: Boolean, reflect: true })
+  open = false;
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.modalRenderRoot = this.createModalRenderRoot(); // Creates modal render root up-front to hook the event listener
+    // Manually hooks the event listeners on the modal render root to make the event names configurable
+    this._hCloseModal = on(
+      this.modalRenderRoot,
+      (this.constructor as typeof DDSImageWithCaption).eventCloseModal,
+      this._handleCloseModal as EventListener
+    );
+  }
+
+  disconnectedCallback() {
+    if (this._hCloseModal) {
+      this._hCloseModal = this._hCloseModal.release();
+    }
+    super.disconnectedCallback();
+  }
+
+  render() {
+    return html`
+      ${this.lightbox
+        ? html`
+            <button class="${prefix}--image-with-caption__image" @click="${this._handleClick}">
+              <dds-image alt="${this.alt}" default-src="${this.defaultSrc}" />
+              <div class="${prefix}--image-with-caption__zoom-button">
+                ${ZoomIn20()}
+              </div>
+            </button>
+          `
+        : html`
+            <dds-image default-src="${this.defaultSrc}" />
+          `}
+      <p class="${prefix}--image__caption">
+        ${this.heading}
+      </p>
+    `;
+  }
+
+  /**
+   * The name of the custom event fired after the modal is closed upon a user gesture.
+   */
+  static get eventCloseModal() {
+    return `${ddsPrefix}-modal-closed`;
+  }
+
+  static styles = styles;
+
+  renderModal() {
+    return html`
+      <dds-modal ?open=${this.open} expressive-size="full-width">
+        <bx-modal-close-button></bx-modal-close-button>
+        <dds-lightbox-image-viewer
+          alt="${this.alt}"
+          default-src="${this.defaultSrc}"
+          description="${this.copy}"
+          title="${this.heading}"
+        >
+        </dds-lightbox-image-viewer>
+      </dds-modal>
+    `;
+  }
```

Comment:
Could you move this up next to `.render()`?

Reply:
👍 @asudoh